pkg/pipeline/resource_pipeline_artifactory_integration.go

```diff
@@ -0,0 +1,138 @@
+package pipeline
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/jfrog/terraform-provider-shared/util"
+)
+
+func pipelineArtifactoryProjectIntegrationResource() *schema.Resource {
+
+	var artifactoryIntegrationSchema = util.MergeSchema(
+		projectIntegrationSchema,
+		map[string]*schema.Schema{
+			"apikey": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Sensitive:    true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				Description:  "ApiKey for Artifactory access",
+			},
+			"user": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				Description:  "User for Artifactory access",
+			},
+			"url": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
We can also check for valid URL. https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2@v2.17.0/helper/validation#IsURLWithScheme

Reply:
done